pkg/app/run.go

```diff
@@ -199,6 +203,57 @@ func run(clicontext *cli.Context) error {
 			}()
 		}
 
+		go func() {
```

Comment:
This part can be put into another function. It's too long.

Reply:
sounds good, i will abstract it